ql/ql/src/queries/style/AlertMessage.ql

```diff
@@ -30,6 +34,14 @@ private AstNode getSelectPart(Select sel, int index) {
             )
           )
         )
+        or
+        // the string is a variable that is assigned in the `where` clause.
+        exists(VarAccess v, ComparisonFormula comp, String str |
+          v = getASubExpression(sel) and
+          loc = v.getLocation() and
+          comp.hasOperands(v.getDeclaration().getAnAccess(), str) and
```

Comment:
I wonder... Can `v.getDeclaration().getAnAccess()` ever be a `String`? If not, it would seem that
```ql
comp.getAnOperand() = v.getDeclaration().getAnAccess() and
comp.getAnOperand() = str
```
would suffice.

Reply:
Yes, that would suffice, and that was also how I had it originally.   
It put it like that to try and improve a join order.  
I'm not sure it worked, but I kept it.